develop

PaulYakow opened the pull request.

PaulYakow committed:
create dev branch & correct workflows files

PaulYakow committed:
+ add user proto
+ add usecase, repo for register/auth user & test for it (must fail some of them)
+ add utils for password hashing, token make & random for tests

PaulYakow committed:
+ change module name in go.mod (begin with "github.com/")
+ add golangci config file
+ add tests, mocks & some interfaces

PaulYakow committed:
+ add comments
+ change lint settings (golangci.yml)
+ server -> add controller, app, config (env + YAML)
+ rename struct for repo, usecase

PaulYakow commented:
Текущая реализация "маршрута" регистрации и авторизации пользователей (на стороне сервера).
Оставшуюся часть планирую выполнять таким же образом:
- соответствующие сущности в _entity_ (DAO/DTO) - пары логин/пароль, данные карт и т.д.
- в _repo_ разнести код ответственный за работу с каждой сущностью по файлам и, скорее всего, интерфейсам (как на данный момент сделано для _IAuthorizationRepo_)
- _usecase_'ы аналогично
- в _controller_'е, из методов grpc будут возвращаться ошибки со статус-кодом из пакета grpc/status

Мысли/замечания (для себя) по дальнейшей структуре частично прописал в коде в виде todo/fixme.

При реализации проекта решил освоить и применить подход TDD - довольно интересно и эффективно, но по-началу долго получалось, т.к. непривычно.

mr-fard4y reviewed (approved):
LGTM

PaulYakow committed:
+ add readme and docs
+ separate repo, usecase and controllers (client & server side) into Auth and Pairs parts
+ add TUI for client

PaulYakow commented:
Разделил код по зонам ответственности (аутентификация, работа с типами данных) - в репозитории, логике и контроллере.
В клиентском TUI конечно тоже можно разделить по частям, но пока так.
Добавил немного описания со схемами.

PaulYakow committed:
+ add missing comments

mr-fard4y commented on internal/server/usecase/repo/repo.go:
```diff
@@ -0,0 +1,63 @@
+    password_hash VARCHAR NOT NULL
+);
+
+CREATE TABLE IF NOT EXISTS public.pairs
```
просто так хранить пароли нет особого смысла

нужно объяснение дальнейшего использования
например,
1) "протухающие" пароли и необходимость их периодического перезадания, через 1/3/6 мес - используем created_at 
2) при кейсе1 проверяем, чтобы пароли не повторялись

иное

mr-fard4y replied:
необходимо переименовать таблицу в более конкретное название, передающее ее назначение

resources
resources.pairs_data
storage.private_data

что-то в этом духе

mr-fard4y commented on internal/client/views/client.go:
```diff
@@ -0,0 +1,234 @@
+
+func (v *View) createUnitsMenu() {
+	v.tui.unitsMenu = tview.NewList().
+		AddItem("Pairs", "show login/password pairs", 'r', func() {
```
можно оставить для тестирования, но для боевого сервера будет излишне
большой шанс компрометации данных, а далее брутфорс с перебором значений

PaulYakow committed:
+ add command (to Makefile) for client's cross-compile
+ add flag to set path to config file (`-c` for client & server)
+ add functionality for view all text & card data in TUI
+ correct table names for data types in DB (move to 'resources' schema)
+ edit README.md, Makefile

PaulYakow commented:
Изменений/доработок в рамках выпускного проекта больше не намечается.
Дальнейшие доработки (раздел "Возможное развитие проекта" и неотмеченные пункты по ходу описания) планируются в качестве pet-проекта после завершения обучения.

mr-fard4y commented on proto/pair.proto:
```diff
@@ -0,0 +1,25 @@
+}
+
+service Pair {
+  rpc GetAll(GetAllPairsRequest) returns (GetAllPairsResponse);
```
возможность только получения данных
не рализована возможность добавления/редактирования/удаления

как подразумевается взаимодействие клиента-сервера при таких кейсах?

mr-fard4y commented on proto/text.proto:
```diff
@@ -0,0 +1,24 @@
+}
+
+service Text {
+  rpc GetAll(GetAllNotesRequest) returns (GetAllNotesResponse);
```
возможность только получения данных
не рализована возможность добавления/редактирования/удаления

как подразумевается взаимодействие клиента-сервера при таких кейсах?

в рамках README нужна аргументация выбора решения

mr-fard4y commented on proto/bank.proto:
```diff
@@ -0,0 +1,26 @@
+}
+
+service Bank {
+  rpc GetAll(GetAllCardsRequest) returns (GetAllCardsResponse);
```
возможность только получения данных
не рализована возможность добавления/редактирования/удаления

как подразумевается взаимодействие клиента-сервера при таких кейсах?

PaulYakow committed:
+ В README.md добавлена "Предполагаемая реализация оставшегося функционала"

mr-fard4y reviewed (approved):
LGTM

thnx for your cooperation
